### ISSUE TITLE
Search config separator option doens't matter

### ISSUE THREAD

**Issue by 63l06ri5** [squidfunk/mkdocs-material#3290]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

I found a bug with the search config separator option - it's not used properly, changing value of this parameter is taking no effect.

Braking change was made in 
7725e555f4866e22de73c1a1b1faf2b0abda23d3
Fixed Japanese segmenter not being executed as part of new tokenizer

There was a move from lunr.tokenizer.separator to this.tokenizer.separator

src\assets\javascripts\integrations\search\_\index.ts
separator set through this.tokenizer.separator

src\assets\javascripts\integrations\search\internal\index.ts:
separator used by lunr.tokenizer.separator

### Expected behaviour

If you have separator: [\s\-\.]+
You can find MainSubProcess word in TestHelper.MainSubProcess 

### Actual behaviour

If you have separator: [\s\-\.]+
You can't find MainSubProcess word in TestHelper.MainSubProcess 

### Steps to reproduce

1. Make md file with text TestHelper.MainSubProcess 
2. Set separator in mkdocs.yml: [\s\-\.]+
3. Build
4. Try to find MainSubProcess

### Package versions

- Python: Python 3.10.0
- MkDocs: mkdocs, version 1.2.3
- Material: 7.3.6+insiders.3.2.0

commit 2c3b4497913f3fdb1822cb50f79de65c7f9cf571 (origin/master, origin/HEAD, master)
Author: squidfunk <martin.donath@squidfunk.com>
Date:   Wed Dec 1 00:46:38 2021 +0100

### Configuration

```yaml
site_name: name
site_dir: public
plugins:
  - search:
      separator: '[\s\-\.]+'
```

### System information

- Operating system: windows 10
- Browser: firefox, chrome

**Comment by squidfunk** [comment]
Thanks for reporting. Note that the separator ` [\s-.]+` means "[all ASCII characters between whitespace and dot](https://www.asciitable.com/)". It should really be `'[\s\-\.]+'` as you posted in your `mkdocs.yml`. Does it happen when you add the backslashes?

**Comment by squidfunk** [comment]
Furthermore, please update to the latest version of Insiders and check whether the issue is gone. You're on 3.2.0 and the latest is 4.2.0.

**Comment by squidfunk** [comment]
Okay, so I'm very confused. You're actually right that the separator sometimes doesn't seem to be used, but I don't yet really understand why that is the case. However, digging into the code, I found _a_ potential issue and it now works again. Interestingly the official documentation worked despite of that change. Well, weird. Fixed in [`5952f9c1f`](https://github.com/squidfunk/mkdocs-material-insiders/commit/5952f9c1f29f1f5189c30b8e29ad5ee4eaf893eb).

**Comment by 63l06ri5** [comment]
I think that is fixed. 

By my understanding before the fix the default separator [\s\\-]+ was used all the way - it's a good default so maybe that's why there is no problem with documentation. And maybe in attempt to clear a confusion i should say that a separator option was set to the local function tokenizer instead of lunr's tokenizer. 

Anyway it's fixed, thanks for that and for a quick response
